### PR TITLE
[RubyGemsIntegration] Mark loaded specs as activated

### PR DESCRIPTION
**Pull request by segiddins**

### CODE REVIEW

**Comment by indirect** [comment]
What effects does this change have? Does it fix a bug?

**Comment by segiddins** [comment]
Not that I can tell, but it does mean that the resolved specs are marked as `activated`, which `Specification#activate` does, and it's possible someone is relying on that instead of using the `loaded_specs` hash.

File: lib/bundler/rubygems_integration.rb [hunk 1]
```diff
@@ -39,7 +39,12 @@ def loaded_specs(name)
     end
 
     def mark_loaded(spec)
-      Gem.loaded_specs[spec.name] = spec
+      name = spec.name
+      if existing = Gem.loaded_specs[name]
+        existing.activated = false
```
Comment (segiddins) [inline lib/bundler/rubygems_integration.rb]: probably needs to make sure that `existing.respond_to? :activated` if we want to merge this :/

Comment (indirect) [inline lib/bundler/rubygems_integration.rb]: Heh, yeah. Pretty sure that `activated` was invented later than some of the RubyGems versions we support.

**Comment by segiddins** [comment]
This seems to have been merged manually.

**Pull request closed without merging**